src/ring_theory/ideals.lean

```diff
@@ -164,6 +165,53 @@ theorem is_coprime_self {x : α} :
   is_coprime x x ↔ is_unit x :=
 by rw [← span_singleton_eq_top]; simp [is_coprime]
 
+/-- This is `is_coprime_def` but specialized to `1`. -/
+lemma is_coprime_def' {α : Type u} [comm_ring α] {x y : α} :
+  is_coprime x y ↔ ∃ (a b : α), a * x + b * y = 1 :=
+⟨λ h, is_coprime_def.1 h 1,
+λ ⟨a, b, h⟩, is_coprime_def.2 $ λ z,
+⟨z * a, z * b, by rw [mul_assoc, mul_assoc, ← mul_add, h, mul_one]⟩⟩
+
+theorem mul_dvd_of_coprime {α : Type u} [comm_ring α] {x y z : α} (H : is_coprime x y)
+  (H1 : x ∣ z) (H2 : y ∣ z) : x * y ∣ z :=
+begin
+  obtain ⟨a, b, h⟩ := is_coprime_def.1 H 1,
+  rw [← mul_one z, ← h, mul_add],
+  apply dvd_add,
+  { rw [mul_comm z, mul_assoc],
+    exact dvd_mul_of_dvd_right (mul_dvd_mul_left _ H2) _ },
+  { rw [mul_comm b, ← mul_assoc],
+    exact dvd_mul_of_dvd_left (mul_dvd_mul_right H1 _) _ }
+end
+
+theorem is_coprime_prod_of_pairwise_coprime {α : Type u} [comm_ring α]
+  {I : Type v} {s : I → α} (hs : pairwise (is_coprime on s)) {t : finset I} {x : I} :
```

Comment:
What is this `on` in `is_coprime on s`?

Reply:
https://github.com/leanprover-community/lean/blob/be6ec5c6d5122823889981b65bd059d865f5746b/library/init/function.lean#L38-L55

```lean
@[reducible] def on_fun (f : β → β → φ) (g : α → β) : α → α → φ :=
λ x y, f (g x) (g y)

infixl  ` on `:2         := on_fun
```

Reply:
TIL